implementations/rust/ockam/ockam_app_lib/src/state/mod.rs

```diff
@@ -581,33 +596,17 @@ impl AppState {
                         invitations
                     },
                     incoming_services: {
-                        let mut incoming_services: Vec<Service> = invitation_state
-                            .accepted
-                            .invitations
+                        let mut incoming_services: Vec<Service> = incoming_services_state
+                            .services
                             .iter()
-                            .filter(|invitation| {
-                                invitation.invitation.owner_email == email
-                                    && invitation.service_access_details.is_some()
-                            })
-                            .map(|invitation| {
-                                let access_details =
-                                    invitation.service_access_details.as_ref().unwrap();
-                                let inlet = invitation_state
-                                    .accepted
-                                    .inlets
-                                    .get(&invitation.invitation.id);
-
-                                Service {
-                                    id: invitation.invitation.id.clone(),
-                                    source_name: access_details
-                                        .service_name()
-                                        .unwrap_or("unknown".to_string()),
-                                    address: inlet.map(|inlet| inlet.socket_addr.ip().to_string()),
-                                    port: inlet.map(|inlet| inlet.socket_addr.port()),
-                                    scheme: None,
-                                    available: inlet.is_some(),
-                                    enabled: inlet.map(|inlet| inlet.enabled).unwrap_or(true),
-                                }
+                            .map(|service| Service {
```

Comment:
Naming the variable `service` to `incoming_service` would be slightly clearer. Otherwise it seems that we are transforming a `Service` into a `Service`. A better alternative IMO would be to have a `Service::from_incoming_service` function to pass to `map`.

Reply:
> Otherwise it seems that we are transforming a Service into a Service

Actually we are.. from one internal representation with the full state to a simplified, C-compatible one.
Maybe adding the whole module would remove some of the confusion, what do you think?

Reply:
Yes, that would help understand that this is indeed transformed data. We can merge and do that if we come back for a second pass.